src/rlang-dev.h

```diff
@@ -0,0 +1,36 @@
+#ifndef VCTRS_RLANG_DEV_H
+#define VCTRS_RLANG_DEV_H
```

Comment:
Is the idea that this is a holding place for ideas that might move to rlang?

Reply:
Exactly, it seems like `struct r_lazy` would be a good addition. I mostly added the file to solve a header deps issue.